src/azul/plugins/utilities.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional
+
+from azul import (
+    reject,
+    require,
+)
+
+
+def unify_string_pair(str_one: Optional[str], str_two: Optional[str]) -> Optional[str]:
```

Comment:
This is sooooo specific to how the deals with units and values in HCA metadata. There is no point in pretending it has general use. It just makes the code harder to read because its void of the context in which its is used. There is also no unification going on. Just concatenation, which is different. 

This should be a method in the HCA metadata plugin. The ElasticsearchService already has a reference to an instance of that plugin, it's just typed to the more general MetadataPlugin type as opposed to the concrete HCA subclass. You may need an assert or a cast to convince PyCharm of the type.